### PR TITLE
fix(turborepo): avoid starting ui on too small terminals

### PR DESCRIPTION
**Pull request by NicholasLYang**
### Description

Avoids starting the UI when the terminal is too small. Also bumps crossterm because apparently we get an incorrect terminal size on older versions.

### Testing Instructions

Tested with `lefthook` on create-turbo

### CODE REVIEW

File: crates/turborepo-ui/src/tui/app.rs [hunk 1]
```diff
@@ -160,6 +159,14 @@ fn poll(input_options: InputOptions, receiver: &AppReceiver, deadline: Instant)
     }
 }
 
+const MIN_HEIGHT: u16 = 10;
```
Comment (NicholasLYang) [inline crates/turborepo-ui/src/tui/app.rs:162]: These are completely arbitrary. Open to suggestions on what they should actually be

Comment (chris-olszewski) [reply crates/turborepo-ui/src/tui/app.rs:162]: I think these are completely reasonable. If I'm accounting for borders correctly 20x10 term would probably lead to a 12x8 terminal pane. Not sure if that's usable, but anything smaller is definitely not usable.

**Review by chris-olszewski — APPROVED** [review]
LGTM

File: crates/turborepo-ui/src/tui/app.rs (hunk 1 above)
Comment (mehulkar) [reply crates/turborepo-ui/src/tui/app.rs:162]: add a comment that it's arbitrary?

**Review by mehulkar — APPROVED** [review]

**Pull request merged**